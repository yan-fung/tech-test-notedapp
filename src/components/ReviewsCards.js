const ReviewsCards = ({ body, date, name, rating, title }) => {
  const formattedDate = date ? date.toDate().toLocaleDateString() : "";
  return (
    <>
      <div>
        <div className="title">{title}</div>
        <div className="rating" data-testid="review_rating">
          {rating}
        </div>
        <div className="date">{formattedDate}</div>
        <div className="name">{name}</div>
        <p className="body">{body}</p>
      </div>
    </>
  );
};

export default ReviewsCards;
