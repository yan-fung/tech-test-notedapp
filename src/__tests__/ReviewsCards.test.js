import { render } from "@testing-library/react";
import ReviewsCards from "../components/ReviewsCards";
import "@testing-library/jest-dom/extend-expect";

describe("ReviewCards", () => {
  const validProps = {
    name: "user1",
    title: "Excellent",
    formattedDate: "2023-07-20",
    rating: "5",
    body: "Amazing app",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ReviewsCards
        name={validProps.name}
        title={validProps.title}
        date={validProps.date}
        rating={validProps.rating}
        body={validProps.body}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId, getByText } = render(
      <ReviewsCards
        name={validProps.name}
        title={validProps.title}
        date={validProps.date}
        rating={validProps.rating}
        body={validProps.body}
      />
    );
    expect(getByText(/Excellent/)).toHaveClass("title");
    expect(getByTestId("review_rating")).toHaveAttribute("class", "rating");
  });
});
