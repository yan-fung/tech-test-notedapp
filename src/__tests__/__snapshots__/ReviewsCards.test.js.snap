// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCards renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="title"
    >
      Excellent
    </div>
    <div
      class="rating"
      data-testid="review_rating"
    >
      5
    </div>
    <div
      class="date"
    />
    <div
      class="name"
    >
      user1
    </div>
    <p
      class="body"
    >
      Amazing app
    </p>
  </div>
</DocumentFragment>
`;
